paparazzi/paparazzi-gradle-plugin/src/test/projects/view-with-lifecycle-usages/src/test/java/app/cash/paparazzi/plugin/test/LifecycleUsageTest.kt

```diff
@@ -0,0 +1,38 @@
+package app.cash.paparazzi.plugin.test
+
+import android.view.View
+import androidx.activity.findViewTreeOnBackPressedDispatcherOwner
+import androidx.activity.OnBackPressedCallback
+import androidx.activity.OnBackPressedDispatcherOwner
+import app.cash.paparazzi.DeviceConfig
+import app.cash.paparazzi.Paparazzi
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+class LifecycleUsageTest {
+  @get:Rule
+  val paparazzi = Paparazzi()
+
+  @Test fun OnBackPressedDispatcherOwner() {
+    val view = View(paparazzi.context)
+    view.doOnAttach {
+      val dispatcherOwner = checkNotNull(view.findViewTreeOnBackPressedDispatcherOwner()) {
+        "Could not find a OnBackPressedDispatcherOwner"
+      }
+      dispatcherOwner.getOnBackPressedDispatcher().addCallback(
+        object : OnBackPressedCallback(/* enabled = */ true) {
+          override fun handleOnBackPressed() = Unit
+        }
+      )
+    }
+    paparazzi.snapshot(view)
+  }
```

Comment:
I'll add similar tests for `LifecycleOwner` and `SavedStateRegistryOwner` as a follow-up once this is approved.

Reply:
sounds good, please add those to this same test in the follow-up.